analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCheckZeroSizeCollection.CSharp10.cs

```diff
@@ -3,11 +3,20 @@
 
 var r = new R();
 
-if (r is R { SomeProperty.Length: >= 0 }) // FN
+if (r is R { SomeProperty.Length: >= 0 }) // Noncompliant
+{
+}
+
+if (r is R { SomeOtherProperty.SomeProperty.LongLength: >= 0 }) // Noncompliant
```

Comment:
Could you please add some tests with nested patterns as well?
For example:
```csharp
using System;
using System.Collections.Generic;

namespace ConsoleApp1
{

    public class Class1
    {

        public void  Check(R r)
        {
            if (r is { Collection: List<R> { Count : > 0 } })
            {
                DoSomething();
            }
        }

        private void DoSomething()
        {
            throw new NotImplementedException();
        }
    }

    public class R
    {
        public readonly List<R> Collection = new();
    }
}
```

Reply:
These cases already exist in the Csharp9 test cases.

Reply:
Nested patterns are already covered. 
Sorry for the noise.